bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java

```diff
@@ -307,12 +307,18 @@ public void attributeChanged(String name, Object value, MqttBrokerConnection con
      */
     public ArrayList<String> getRetainedTopics() {
         ArrayList<String> topics = new ArrayList<String>();
-        Map<String, Object> map = this.attributes.asMap();
-        topics.addAll(map.keySet().stream().map(
-                a -> {return String.format("%s/$%s", this.propertyID, a);}).collect(Collectors.toList()));
 
-        if(Boolean.parseBoolean(map.get("retained").toString()))
-            topics.add(this.propertyID);
+        topics.addAll(Stream.of(this.attributes.getClass().getDeclaredFields()).map(
+            f -> {return String.format("%s/$%s", this.propertyID, f.getName());}).collect(Collectors.toList()));
+
+        try {
+		    if(attributes.getClass().getDeclaredField("retained").getBoolean(attributes))
+                topics.add(this.propertyID);
+        } catch (NoSuchFieldException e) {
```

Comment:
Please name ignored exception variables "ignored" (for the static checker) and comment why those can be ignored without a log line.

Reply:
Done! (Thanks for your patience, a learn a lot from your suggestions 👍)